scripts/fast-reboot

```diff
@@ -156,6 +156,55 @@ function request_pre_shutdown()
     }
 }
 
+function recover_issu_bank_file_instruction()
+{
+    error "To recover (${ISSU_BANK_FILE}) file, do the following:"
+    error "$ docker exec -it syncd sx_api_dbg_generate_dump.py"
+    error "$ docker exec -it syncd cat /tmp/sdkdump | grep 'ISSU Bank'"
+    error "Command above will print the VALUE of ISSU BANK - 0 or 1, use this VALUE in the next command"
+    error "$ printf VALUE > /host/warmboot/issu_bank.txt"
+}
+
+function check_issu_bank_file()
+{
+    ISSU_BANK_FILE=/host/warmboot/issu_bank.txt
+    MLNX_ISSU_BANK_BROKEN_BEFORE_PRE_SHUTDOWN=102
+    MLNX_ISSU_BANK_BROKEN_AFTER_PRE_SHUTDOWN=103
+
+    if [[ ! -s "$ISSU_BANK_FILE" ]]; then
+        error "(${ISSU_BANK_FILE}) does NOT exist or empty ..."
+        recover_issu_bank_file_instruction
+	if [[ "$1" = true ]]; then
+            exit "${MLNX_ISSU_BANK_BROKEN_BEFORE_PRE_SHUTDOWN}"
+	else
+            exit "${MLNX_ISSU_BANK_BROKEN_AFTER_PRE_SHUTDOWN}"
```

Comment:
Sorry I missed this issue in my second round of review.

Exit here will cause the fast/warm reboot to abort. I don't think this is what we want here. Because at the time when this check happens, we've already shutdown some critical services, e.g. BGP, RADV. Depending on if pre-shutdown is called, we could also lose syncd. It is unfortunate that there was an issue. But I think at this point, we would have to proceed with reboot and take the consequence. The logic is that aborting reboot sequence will require human intervention to recover. Continue rebooting will cause data plane disruption in the future. That sucks, but that would self heal. Right?

I think here the best we can do is to generate a log message so that we have log to look into after reboot is done and learn what happened when.

Reply:
> Sorry I missed this issue in my second round of review.
> 
> Exit here will cause the fast/warm reboot to abort. I don't think this is what we want here. Because at the time when this check happens, we've already shutdown some critical services, e.g. BGP, RADV. Depending on if pre-shutdown is called, we could also lose syncd. It is unfortunate that there was an issue. But I think at this point, we would have to proceed with reboot and take the consequence. The logic is that aborting reboot sequence will require human intervention to recover. Continue rebooting will cause data plane disruption in the future. That sucks, but that would self heal. Right?
> 
> I think here the best we can do is to generate a log message so that we have log to look into after reboot is done and learn what happened when.

Fixed